packages/amplify_flutter/lib/src/utils/equatable.dart

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import 'package:collection/collection.dart';
+
+/// Provides overrides for [==] and [hashCode] for consistent equality and
+/// hashing.
+mixin AWSEquatable<T extends AWSEquatable<T>> on Object {
+  /// The object's properties which should be equated and hashed.
+  List<Object?> get props;
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is T &&
+          const DeepCollectionEquality.unordered().equals(props, other.props);
```

Comment:
Shouldn't this be ordered? With it unordered, the following evaluates to true

```dart
AmplifyConfig(userAgent: 'foo', version: 'bar') == AmplifyConfig(userAgent: 'bar', version: 'foo') 
```

A unit test on this would be nice.

Reply:
No, unordered only applies to lists/iterables of things. It just means `['a', 'b', 'c'] == ['b', 'c', 'a']`.

Reply:
Oh I see what you're saying - yeah, I can add some tests.

Reply:
Good catch - should not be unordered. Fixed!